sdk/storage/Azure.Storage.DataMovement/tests/PauseResumeTransferMockedTests.cs

```diff
@@ -435,11 +551,84 @@ public async Task PauseDuringChunkProcessing_ItemTransfer(
+            Assert.That(inProgressJobsCount, Is.EqualTo(items / 2), "Error: all jobs should be in InProgress state after Job Processing on resume");
+            Assert.That(pausedJobPartsCount, Is.EqualTo(items / 2));
+            Assert.That(partsProcessor.ItemsInQueue, Is.EqualTo(items / 2), "Error in job part processor on resume");
+        }
+        else
+        {
+            Assert.That(inProgressJobsCount, Is.EqualTo(items), "Error: all jobs should be in InProgress state after Job Processing on resume");
+            Assert.That(pausedJobPartsCount, Is.EqualTo(items));
+            Assert.That(partsProcessor.ItemsInQueue, Is.EqualTo(items), "Error in job part processor on resume");
+        }
+
+        // process job parts on resume
+        await partsProcessor.StepAll();
+
+        await Task.Delay(50);
+        int inProgressJobPartsCount = GetJobPartsStateCount(resumedTransfers, checkpointer)[DataTransferState.InProgress];
+        if (pauseLocation == PauseLocation.PauseProcessHalfway)
+        {
+            // the count for all job parts for PauseProcessHalfway is (items / 2) since half have already completed transfer
+            Assert.That(inProgressJobPartsCount, Is.EqualTo(items / 2), "Error: all job parts should be in InProgress state after Part Processing on resume");
+            Assert.That(chunksProcessor.ItemsInQueue, Is.EqualTo(numChunks / 2), "Error in chunk processor on resume");
+        }
+        else
+        {
+            Assert.That(inProgressJobPartsCount, Is.EqualTo(items), "Error: all job parts should be in InProgress state after Part Processing on resume");
+            Assert.That(chunksProcessor.ItemsInQueue, Is.EqualTo(numChunks), "Error in chunk processor on resume");
+        }
+
+        // process chunks on resume
+        await chunksProcessor.StepAll();
+
+        // Assert that all jobs and job parts are completed successfully
+        await Task.Delay(50);
+        int completedJobsCount2 = GetJobsStateCount(transfers, checkpointer)[DataTransferState.Completed];
+        int completedPartsCount2 = GetJobPartsStateCount(transfers, checkpointer)[DataTransferState.Completed];
+        Assert.That(completedJobsCount2, Is.EqualTo(items), "Error in transitioning all jobs to Completed state");
+        Assert.That(completedPartsCount2, Is.EqualTo(items), "Error in transitioning all job parts to Completed state");
+        foreach (DataTransfer resumedTransfer in resumedTransfers)
+        {
+            Assert.That(resumedTransfer.HasCompleted);
+            Job job = checkpointer.Jobs[resumedTransfer.Id];
+            Assert.That(job.Status.State, Is.EqualTo(DataTransferState.Completed), "Error in transitioning all jobs to Completed state");
+            JobPart jobPart = job.Parts.First().Value;
+            Assert.That(jobPart.Status.State, Is.EqualTo(DataTransferState.Completed), "Error in transitioning all job parts to Completed state");
+        }
```

Comment:
Looks like this part of the code is the same in the other places you added. Add a private function that it calls instead so we can avoid some code duplication. Also this would ensure in the future when we need to add the same assertions, they're all added for all the tests that call it.

Reply:
@amnguye thanks for the review. So the asserts are very similar across test cases but there are little differences and nuances between each of them where it makes it hard to generalize into one helper method. 

Like for example, for pausing & resuming during job processing, you end up with some jobs that don't have job parts when the resume begins. This is very different than pausing and resuming during parts and chunk processing where you are guaranteed that each job will have all job parts which is why the asserts will be a little different. Another example is if you pause midway through chunk processing, half of the chunks will have finished processing already so only the jobs that didn't complete will be resumed (which is very different than in the other tests where all jobs are guaranteed to be resumed).